### ISSUE TITLE
Where's the loss1 and loss2 in GoogLeNet

### ISSUE THREAD

**Issue by haqishen** [vlfeat/matconvnet#353]
Hi,

  I've downloaded the pre-trained model GoogLeNet and learning about it.

  There actually should be 3 softmax layers names 'loss1', 'loss2', 'loss3' in this model (according to https://www.dropbox.com/s/08civrafi8vlqji/googlenet.svg?dl=1) but I can only find one which is the last layer of this model (loss3).

  I tried to track the data flow and find that no layers use 'cls1_fc2' and 'cls2_fc2' as inputs, which means those layers names 'cls1' and 'cls2' may have done no job in this model.

  Thanks a lot if anyone can give some ideas.

best,
Ha

**Comment by xingchao92** [comment]
In my opinion, if you want to fine-tune the GoogLeNet, you can add the Loss Layer named 'loss1' or 'loss2' with the input of 'cls1_fc2' or 'cls2_fc2' .

**Comment by yuzhong84** [comment]
I think loss1 and loss2 are used in training only to improve the gradient flow in backpropagation. Once the model trained, they are removed as only layer from loss3 will be used for classification.

**Comment by vedaldi** [comment]
Hi, yes, this is correct. Note that the model available on the website is a port from the Princeton Caffe implementation of GoogleNet. We are working on our own version (more specifically, inception v3) but are still trying to figure out the tuning.
